src/app/clusters/level-control/level-control.cpp

```diff
@@ -363,18 +369,68 @@ static void reallyUpdateCoupledColorTemp(EndpointId endpoint)
 }
 #endif // IGNORE_LEVEL_CONTROL_CLUSTER_OPTIONS && MATTER_DM_PLUGIN_COLOR_CONTROL_SERVER_TEMP
 
+/*
+ * @brief
+ * This function is used to update the current level attribute
+ * while respecting it's defined quiet reporting quality:
+ * The attribute will be reported:
+ * - At most once per second, or
+ * - At the start of the movement/transition, or
+ * - At the end of the movement/transition, or
+ * - When it changes from null to any other value and vice versa.
+ *
+ * @param endpoint: endpoint on which the currentLevel attribute must be updated.
+ * @param state: LevelControlState struct of this given endpoint.
+ * @param newValue: Value to update the attribute with
+ * @param isStartOrEndOfTransition: Boolean that indicate whether the update is occuring at the start or end of a level transition
+ * @return Success in setting the attribute value or the IM error code for the failure.
+ */
+static Status SetCurrentLevelQuietReport(EndpointId endpoint, EmberAfLevelControlState * state,
+                                         DataModel::Nullable<uint8_t> newValue, bool isStartOrEndOfTransition)
+{
+    AttributeDirtyState dirtyState;
+    MarkAttributeDirty markDirty = MarkAttributeDirty::kNo;
+    auto now                     = System::SystemClock().GetMonotonicTimestamp();
+
+    if (isStartOrEndOfTransition)
+    {
+        // At the start or end of the movement/transition we must report
+        auto predicate = [](const decltype(state->quietCurrentLevel)::SufficientChangePredicateCandidate &) -> bool {
+            return true;
+        };
+        dirtyState = state->quietCurrentLevel.SetValue(newValue, now, predicate);
+    }
+    else
+    {
+        // During transtions, reports should be at most once per second
+        System::Clock::Milliseconds64 reportInterval =
```

Comment:
We should consider, in a follow-up to make this longer (e.g. 5 seconds, as per one of my recommendations in spec PR). This is strictly better than today, so we can go with this for now.

Reply:
Agreed, I applied what you suggested before,  for a long transition we will report at max Transition time / 4 so anything longer than 4 seconds, we would start seeing benefits.